### ISSUE TITLE
Nested group icon does not change colors

### ISSUE THREAD

**Issue by tboyce1** [home-assistant/home-assistant-polymer#186]
Group icons do not change colors when all the entities in the group are "on". I originally reported this [here](https://github.com/home-assistant/home-assistant/issues/4923) but after looking into things a bit more, I think it belongs here.

```
group:
  living_room_lights:
    - light.living_room_light_1
    - light.living_room_light_2

  lights:
    - light.basement_lights
    - group.living_room_lights
```

![image](https://cloud.githubusercontent.com/assets/1326315/21213377/8cea56ec-c259-11e6-8928-a58fd4f1498f.png)

**Comment by tboyce1** [comment]
I believe https://github.com/tboyce1/home-assistant-polymer/commit/e9e6bfac43ffc4906c2ca76b1a52ee4216d67d81 is what needs to be done, but I could be wrong and perhaps there is some reason group icons do not change colors.

**Comment by balloob** [comment]
The reason it doesn't turn on is because 'on' for a group doesn't always makes sense - for example a group of binary sensors is also on.

**Comment by tboyce1** [comment]
I disagree that a group of binary sensors turning 'on' doesn't make sense. It should be the same behavior as switches - if any of the sensors are 'on', then the group is considered 'on'. This way, you could have all your sensors that are normally 'off' in a group on the frontend and tell if any of them turn 'on' by just looking at the group indicator. Then you can click on that group to see which specific one is 'on' if you want. In fact, this is how already it works for the group's state, it just doesn't reflect that in the frontend.

    trigger:
      platform: state
      entity_id: group.some_binary_sensors
      state: 'on'

This will trigger if any of the binary sensors in the group turns on. Also, if you create a group of binary sensors and nest it within another group, it currently does show a switch by default and that switch does turn on/off when any of the binary sensors do. I don't think it should have a switch in this case since you can't actually modify them via the switch (which can be accomplished via "control: hidden"), but I think it should still have the icon show the group as 'on' to reflect the actual state of the group.

**Comment by tboyce1** [comment]
I didn't a little more testing with groups using sensor, binary sensor, camera, climate, sun, script, other groups, zone, weather, and device tracker. In all cases the group's reported state behaves exactly as I would suspect: 'uknown' when there are no entities with 'on'/'off' states and 'on'/'off' when there are. The only exception to this seemed to be when there was a device tracker in the group, in which case the group would show as 'home'/'not_home' if the device tracker was listed before any entities with an 'on'/'off' state, which again is exactly what I would expect.

So basically, the state the group reports seems to work perfectly. The only issue is that the group's icon doesn't change colors to indicate when the group's state is 'on'.

**Comment by balloob** [comment]
Your tests are correct. I should have clarified my statement, I meant to say that it doesn't make sense to color the icon because only for lights/switches can "on" mean that a light is shining.

**Comment by tboyce1** [comment]
That makes more sense but binary sensors and sun currently have colored icons when 'on' so there is some inconsistency there. Also, a switch being on doesn't always mean there is a light either. It could be a switch for a fan or something as well. There are also cases where a binary sensor could mean there is a light shining, e.g. a light sensor.

Personally I would still like this change. So either I'll have to run the frontend in development mode or perhaps this could be made configurable similar to whether the group shows a switch?

**Comment by tboyce1** [comment]
At the very least it would be nice to have a light group behave this way. I'm alright with other groups not coloring the icon. https://community.home-assistant.io/t/three-smart-bulbs-in-a-group-can-i-get-the-default-dynamic-bulb-icon-to-work/10318 also suggests taking it a step further and using the color of the lights themselves but there would need to be a decision on which color to use if they're not all the same (I imagine just the first one is a good enough place to start).

**Comment by andrey-git** [comment]
@balloob
I think changing the color of the light is a good idea. Any indication is better than no indication.

 @tboyce1 
That said, starting from 0.38 you will be able to write your own state-card-group

**Comment by tboyce1** [comment]
Yeah, after looking at things a bit more, updating the group icon for lights could get a bit tricky since colored lights change the icon based on the color of the light, not just whether it's 'on' or 'off'.

Maybe it would be nice to just have a "template" or "lightgroup" light that can be used to essentially treat a group of lights as a single light, keeping their state and color in sync so it's easier to handle the icon.

Fortunately, I have Hue lights in the ones I want to group so I've just been using groups on the bridge instead. I'd still be interested in seeing it for groups of binary sensors, etc., but like you said that can be accomplished with a custom state card. And really it should be possible to write the custom card in a way that it would work for everyone who wants this behavior.

**Comment by flyingartichoke** [comment]
I'd also like to see the icon color change in this case (regardless of whether it's a light bulb, switch, or other icon).  The change in color is a quick visual indicator that "something" is on... I can dive down deeper if I need to know what.

**Comment by zardoz-net** [comment]
> Maybe it would be nice to just have a "template" or "lightgroup" light that can be used to essentially treat a group of lights as a single light so it's easier to handle the icon.

I have followed your idea and it works out of the box:
```
group:
    regal:
    name: Regal
    entities:
      - switch.abc001234
      - switch.abc012345
      - switch.abc123456

switch copy:
  - platform: template
    switches:
      regallicht:
        friendly_name: Regallicht
        value_template: "{{ is_state('group.regal', 'on') }}"
        turn_on:
          service: switch.turn_on
          entity_id: group.regal
        turn_off:
          service: switch.turn_off
          entity_id: group.regal
```
Don't forget to hide the individual switches in HomeBridge and Emulated_Hue, if you use it, e.g.

```
customize:
    switch.abc001234:
      homebridge_hidden: true
      emulated_hue: false
    switch.abc012345:
      homebridge_hidden: true
      emulated_hue: false
    switch.abc123456:
      homebridge_hidden: true
      emulated_hue: false
```

**Comment by NovapaX** [comment]
this issue can be closed because we now (soon) have a light.group component.
https://github.com/home-assistant/home-assistant/pull/12229#issuecomment-370002908
